File: lending-database/src/main/java/info/armado/ausleihe/database/converter/BarcodeConverter.java
/**
 * 
 */
package info.armado.ausleihe.database.converter;

import info.armado.ausleihe.database.barcode.Barcode;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

/**
 * @author Arndt
 *
 */
@Converter(autoApply=true)
public class BarcodeConverter implements AttributeConverter<Barcode, String> {

	@Override
	public String convertToDatabaseColumn(Barcode attribute) {
		return attribute.toString();
	}

	@Override
	public Barcode convertToEntityAttribute(String dbData) {
		return new Barcode(dbData);
	}

}

File: lending-database/src/main/scala/info/armado/ausleihe/database/converter/BarcodeConverter.scala
package info.armado.ausleihe.database.converter

import javax.persistence.{AttributeConverter, Converter}

import info.armado.ausleihe.database.barcode.Barcode

/**
  * A converter used to convert [[Barcode]] instances to [[String]]s to be able to persist them inside a database.
  *
  * @author Marc Arndt
  * @since 24.06.17
  */
@Converter(autoApply = true)
class BarcodeConverter extends AttributeConverter[Barcode, String] {
  override def convertToEntityAttribute(databaseValue: String): Barcode = Barcode(databaseValue)

  override def convertToDatabaseColumn(entityValue: Barcode): String = entityValue.toString
}
